# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/storage/SessionStorageManager.cpp
# Commit: 2616040b2fc8
# Full Hash: 2616040b2fc8813de6f590e5f9cb08386a3991a8
# Author: Yaron Tausky <ytausky@mozilla.com>
# Date: 2020-01-08 21:56:06
# Description:
#   Bug 1603976 - Warn when getting session storage data for discarded browsing context r=dom-workers-and-storage-reviewers,sg,janv
#   
#   Under some (yet) unclear conditions it's possible for a content
#   process to send session storage data for a browsing context that's
#   already been discarded. This leads to an assertion failure when
# ==============================================================================

diff -r 7468eacbeec5 -r 2616040b2fc8 dom/storage/SessionStorageManager.cpp
--- a/dom/storage/SessionStorageManager.cpp	Wed Jan 08 15:30:44 2020 +0000
+++ b/dom/storage/SessionStorageManager.cpp	Wed Jan 08 15:38:34 2020 +0000
@@ -328,7 +328,7 @@
   nsTArray<KeyValuePair> sessionData =
       aCache->SerializeData(SessionStorageCache::eSessionSetType);
   Unused << aActor->SendSessionStorageData(
-      mBrowsingContext, nsCString{aOriginAttrs}, nsCString{aOriginKey},
+      mBrowsingContext->Id(), nsCString{aOriginAttrs}, nsCString{aOriginKey},
       defaultData, sessionData);
 }
 
